### ISSUE TITLE
Problems with video (camera access, switch cameras, image orientation)

### ISSUE THREAD

**Issue by Gabriele69** [bittner/appear-in#1]
Hi, 
web application for Ubuntu Phone does not allow access to the camera .

Gabriele

**Comment by bittner** [comment]
Thank you for reporting this issue. There is a new version uploaded to the Store, which tries to address the problem, but it didn't pass the automatic review process. A manual review by Canonical is in progress. Please be patient.

## Temporary Work-around

In the meantime you can enjoy the glorious [appear.in](https://appear.in/) service simply in your Ubuntu phone's web browser. ([test-drive](https://appear.in/test-drive))

**Note:** You'll probably get your **rear camera** working by default in the browser instead of the front camera that you most probably will want to use. You can _switch cameras_ in the web browser Settings menu (3 stripes in the right upper corner):

> Settings -> Privacy & Permissions -> Camera & Microphone

Afterwards, _reload the web page_ in your mobile browser, then the front camera will be active.

**Comment by Gabriele69** [comment]
Hi Bittner;

Yes, I already tried using the web service and change the Camera settings. But it remains the rear camera even if the browser is set on the front camera.

in any case expect the new version of your app , just the Canonical team will give its ok .

Thank you.

GM

**Comment by bittner** [comment]
You're right. I forgot to mention, after changing the camera settings you need to reload the web page in the browser. Then it works. (I'll update the instructions above, thanks for the hint!)

**Comment by bittner** [comment]
**Update:** Version 0.2 has been released to the Ubuntu Store. Unfortunately, the changes do not fix the video issue. `camera` access is [granted via policy_groups](https://developer.ubuntu.com/en/start/platform/guides/app-confinement/#Usage-1), but no trust-store prompt pops up. It might be we're facing a bug of the `webapp-container`.

Everyone interested in details can follow the [discussion on the mailing list](https://lists.launchpad.net/ubuntu-phone/msg18658.html).

**Comment by bittner** [comment]
**Update:** The issue is related to the `webapp-container` on Ubuntu Touch, a binary shipped with the `webbrowser-app` (oxide). Implementation needs to be completed by Canonical. See [bug 1554202](https://bugs.launchpad.net/ubuntu/+source/webbrowser-app/+bug/1554202) on Launchpad. The camera switching issue is tracked by bug <strike>[1554244](https://bugs.launchpad.net/ubuntu/+source/webbrowser-app/+bug/1554244)</strike> > <strike>[1554603](https://bugs.launchpad.net/oxide/+bug/1554603)</strike> > [1535818](https://bugs.launchpad.net/oxide/+bug/1535818) on Launchpad.

**Comment by bittner** [comment]
**Update:** Medium-good news, still no final solution for the problem. OTA-10 has been released, and Video is now available in the webapp. However, there are still two issues:
- The rear camera seems to be used by default, and there is (still) no way to switch cameras in the webapp. (This feature is available in a sort of clumsy way in the webbrowser only for the moment.)
- <strike>The video image is mirrored for some reason (at least on my bq E5). This seems to be a general problem of the changes introduced to the `webbrowser` app</strike>. (See the [mailing list](https://lists.launchpad.net/ubuntu-phone/msg19497.html) why this is not a bug.)

<strike>I'll report the latter bug to Canonical</strike>, a fix for the former has been promised for the upcoming update (OTA-11).

**Comment by bittner** [comment]
**News:** OTA-11 will deliver an [active front camera by default](https://bugs.launchpad.net/ubuntu/+source/webbrowser-app/+bug/1563398) in the `webbrowser-app` (and the `webapp-container`).

**Comment by bittner** [comment]
**Update:** OTA-11 is out, and unfortunately the promises did not hold. There is no font camera active by default. A great pity. If you want this to change please comment on the [related bug on Launchpad](https://bugs.launchpad.net/ubuntu/+source/webbrowser-app/+bug/1563398). Canonical should start to hear voices from their user base. They may not move otherwise.

**Comment by bittner** [comment]
**Update:** Good news! Looks like we're finally getting a working appear.in app for Ubuntu Touch:
- [Front-facing camera by default](https://lists.launchpad.net/ubuntu-phone/msg21541.html)
- <strike>[New parameter `--no-australia-mode`](https://lists.launchpad.net/ubuntu-phone/msg21542.html)</strike>

OTA-12 is [delayed until next week](https://lists.launchpad.net/ubuntu-phone/msg21640.html) though, but I'll be releasing the necessary changes before that as the additional parameter doesn't seem to affect the startup of the current (OTA-11) webapp-container.

**Comment by bittner** [comment]
**Update:** Bad news. OTA-12 fixed the default camera, but the orientation of the video image is now upside-down for both portrait _and_ landscape. One thing improved, one thing made worse. Makes the app still unusable as a whole. (The switch mentioned above was a German joke on the mailing list.)

Even worse in total, now you can't use appear.in in the phone's browser, because that behaves exactly the same. Now there is no need to go into the bowser settings anymore to switch the camera in use, but worse than before, there is no way to fix the upside-down video orientation at all, e.g. with a settings option.  :-1: 

Keep your fingers crossed for OTA-13, though it doesn't look like we'll get this fixed as it [goes into feature freeze](https://lists.launchpad.net/ubuntu-phone/msg21993.html) this Friday. The related issue on LaunchPad is [bug 1535820](https://bugs.launchpad.net/oxide/+bug/1535820). Please cast your vote there if this bug affects you!

**Comment by bittner** [comment]
**Update:** Bad news again. OTA-13 is out, and no change to the upside-down image the phone camera provides to the app (err, the `webapp-container` displays, really). It's all up to Canonical.

**Comment by Gabriele69** [comment]
Hello Peter,
I wanted to inform you that on my Acquaris E4.5. Appear works perfectly.
Launching your app, you properly activate the front camera, and the 
image is upright (... and I don't live in Australia! :-))

Il 21/09/2016 17:34, Peter Bittner ha scritto:

**Comment by bittner** [comment]
Thank you for your update, @GabrieleMartelli. This confirms what Canonical's David Barth says in [comment 5 of the Launchpad bug](https://bugs.launchpad.net/oxide/+bug/1535820/comments/5).

Would be interesting to know which other devices are now still affected. The [bq Aquaris E5](https://www.bq.com/en/support/aquaris-e5-ubuntu-edition/support-sheet) is one of them. In any case, this must be fixed in Oxide, the Ubuntu Touch web browser, or a related library.
